src/lib.rs

```diff
@@ -60,12 +54,22 @@ impl Firebase {
     ///   will be returned.
     /// - If a url cannot be parsed into a valid url then a ```Err(ParseError::Parser(url::ParseError)```
     ///   will be returned.
-    pub fn from_url(url: &Url) -> Result<Self, ParseError> {
-        let url = url.clone();
+    pub fn from_url(mut url: Url) -> Result<Self, ParseError> {
         try!( unwrap_path(&url) );
 
+        if url.scheme() != "https" {
+            return Err(ParseError::UrlIsNotHTTPS);
+        }
+
+        {
+            // Append initial .json
+            let mut last = url.path_segments().expect("last segment").last().unwrap().to_string();
+            let mut path = url.path_segments_mut().expect("path segments");
```

Comment:
Yes, it is a valid FB url.

Michael Eden <notifications@github.com> schrieb am Do., 2. Juni 2016, 16:11:

> In src/lib.rs
> <https://github.com/davidrhyswhite/rust-firebase/pull/5#discussion_r65545882>
> :
>
> >          try!( unwrap_path(&url) );
> >
> > +        if url.scheme() != "https" {
> > +            return Err(ParseError::UrlIsNotHTTPS);
> > +        }
> > +
> > +        {
> > +            // Append initial .json
> > +            let mut last = url.path_segments().expect("last segment").last().unwrap().to_string();
> > +            let mut path = url.path_segments_mut().expect("path segments");
>
> This is meant to be the Firebase root, what happens if I type in
> myapp.firebase.com ? is https://myapp.firebase.com/.json a valid firebase
> url?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/davidrhyswhite/rust-firebase/pull/5/files/1bf7965838a89e96743f31766c32b0e7a22493de#r65545882>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAmW3fM_t2oAXUPKT8my4ifE9wvkx_Inks5qHuQagaJpZM4IkvG7>
> .
>